### PR TITLE
Add support for Flutter packages.

### PR DESCRIPTION
**Pull request by matanlurey**
For now, treat `sdk:` as ~ the same thing as `path:`.

This will break down once Flutter is published another way, but allows `build_runner` to start working without crashing when running within a Flutter package. We will (eventually) need to figure out additional things, like how to use the right analyzer to recognize `dart:ui`, but this is a start.

Also ran `dartfmt`, and added a better error message when a dependency is not found (not NPE).

Closes https://github.com/dart-lang/build/issues/211.

### CODE REVIEW

**Comment by natebosch** [comment]
LGTM but Jake is more familiar

**Comment by eseidelGoogle** [comment]
I don't really have any context here.  Looks reasonable.  If you need stronger technical consideration consider asking cbracken, devoncarew or abarth.

File: build_runner/lib/src/package_graph/package_graph.dart [hunk 1]
```diff
@@ -92,7 +92,11 @@ class PackageGraph {
       deps.forEach((name, source) {
         var dep = nodes[name];
         if (dep == null) {
-          var pubspec = _pubspecForUri(packageLocations[name]);
+          var uri = packageLocations[name];
```
Comment (jakemac53) [inline build_runner/lib/src/package_graph/package_graph.dart]: nit: move this error detection to the `_pubspecForUri` method?

File: build_runner/lib/src/package_graph/package_graph.dart [hunk 2]
```diff
@@ -167,6 +171,7 @@ PackageDependencyType _dependencyType(source) {
       case 'hosted':
         return PackageDependencyType.hosted;
       case 'path':
+      case 'sdk': // Until Flutter supports another type, assume same as path.
```
Comment (jakemac53) [inline build_runner/lib/src/package_graph/package_graph.dart]: I have no context here, is it actually equivalent to a path dependency? Or is it coming from pub?

Comment (matanlurey) [inline build_runner/lib/src/package_graph/package_graph.dart]: That is my understanding. Maybe @cbracken or @yjbanov or @devoncarew know better.

**Comment by matanlurey** [comment]
@jakemac53 and @natebosch - OK with merging?

**Comment by kevmoo** [comment]
@natebosch ?

File: build_runner/lib/src/package_graph/package_graph.dart (hunk 2 above)
Comment (abarth) [inline build_runner/lib/src/package_graph/package_graph.dart]: It means that it comes from the Flutter SDK, which the developer downloads.

**Pull request merged**